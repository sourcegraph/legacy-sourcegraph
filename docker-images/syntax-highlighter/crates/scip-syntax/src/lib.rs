pub mod evaluate;
pub mod index;
pub mod io;
pub mod progress;
pub mod scip_strict;
