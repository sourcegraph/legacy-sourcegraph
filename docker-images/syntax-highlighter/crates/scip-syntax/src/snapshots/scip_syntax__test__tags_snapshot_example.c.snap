---
source: crates/scip-syntax/src/lib.rs
expression: "String::from_utf8_lossy(buf_writer.buffer())"
---
{"_type":"tag","name":"age","path":"example.c","language":"c","line":31,"kind":"field","scope":null}
{"_type":"tag","name":"hobby","path":"example.c","language":"c","line":31,"kind":"field","scope":"age"}
{"_type":"tag","name":"object.age","path":"example.c","language":"c","line":37,"kind":"field","scope":null}
{"_type":"tag","name":"object.value","path":"example.c","language":"c","line":37,"kind":"field","scope":"object.age"}
{"_type":"tag","name":"object.name","path":"example.c","language":"c","line":37,"kind":"field","scope":"object.age.object.value"}
{"_type":"tag","name":"object2.name","path":"example.c","language":"c","line":43,"kind":"field","scope":null}
{"_type":"tag","name":"connection.url","path":"example.c","language":"c","line":47,"kind":"field","scope":null}
{"_type":"tag","name":"connection.buffer","path":"example.c","language":"c","line":47,"kind":"field","scope":"connection.url"}
{"_type":"tag","name":"connection.bufsize","path":"example.c","language":"c","line":47,"kind":"field","scope":"connection.url.connection.buffer"}
{"_type":"tag","name":"connection.fd","path":"example.c","language":"c","line":47,"kind":"field","scope":"connection.url.connection.buffer.connection.bufsize"}
{"_type":"tag","name":"connection.complete","path":"example.c","language":"c","line":47,"kind":"field","scope":"connection.url.connection.buffer.connection.bufsize.connection.fd"}
{"_type":"tag","name":"computer.cores","path":"example.c","language":"c","line":57,"kind":"field","scope":null}
{"_type":"tag","name":"number","path":"example.c","language":"c","line":61,"kind":"field","scope":null}
{"_type":"tag","name":"hasNestedStruct.b","path":"example.c","language":"c","line":65,"kind":"field","scope":null}
{"_type":"tag","name":"isNestedStruct.y","path":"example.c","language":"c","line":66,"kind":"field","scope":"hasNestedStruct.b"}
{"_type":"tag","name":"isNestedStruct.x","path":"example.c","language":"c","line":66,"kind":"field","scope":"hasNestedStruct.b.isNestedStruct.y"}
{"_type":"tag","name":"isNestedStruct","path":"example.c","language":"c","line":66,"kind":"struct","scope":"hasNestedStruct.b"}
{"_type":"tag","name":"main","path":"example.c","language":"c","line":112,"kind":"function","scope":null}
{"_type":"tag","name":"returns_int","path":"example.c","language":"c","line":110,"kind":"function","scope":null}
{"_type":"tag","name":"returns_int","path":"example.c","language":"c","line":108,"kind":"function","scope":null}
{"_type":"tag","name":"free_connection","path":"example.c","language":"c","line":95,"kind":"function","scope":null}
{"_type":"tag","name":"new_connection","path":"example.c","language":"c","line":75,"kind":"function","scope":null}
{"_type":"tag","name":"new_connection","path":"example.c","language":"c","line":73,"kind":"function","scope":null}
{"_type":"tag","name":"hasNestedStruct","path":"example.c","language":"c","line":65,"kind":"struct","scope":null}
{"_type":"tag","name":"NoName","path":"example.c","language":"c","line":63,"kind":"typeAlias","scope":null}
{"_type":"tag","name":"Computer","path":"example.c","language":"c","line":59,"kind":"typeAlias","scope":null}
{"_type":"tag","name":"computer","path":"example.c","language":"c","line":57,"kind":"struct","scope":null}
{"_type":"tag","name":"Connection","path":"example.c","language":"c","line":55,"kind":"typeAlias","scope":null}
{"_type":"tag","name":"connection","path":"example.c","language":"c","line":47,"kind":"struct","scope":null}
{"_type":"tag","name":"obj4","path":"example.c","language":"c","line":45,"kind":"constant","scope":null}
{"_type":"tag","name":"object2","path":"example.c","language":"c","line":43,"kind":"union","scope":null}
{"_type":"tag","name":"obj3","path":"example.c","language":"c","line":41,"kind":"variable","scope":null}
{"_type":"tag","name":"obj2","path":"example.c","language":"c","line":41,"kind":"variable","scope":null}
{"_type":"tag","name":"obj1","path":"example.c","language":"c","line":41,"kind":"variable","scope":null}
{"_type":"tag","name":"object","path":"example.c","language":"c","line":37,"kind":"union","scope":null}
{"_type":"tag","name":"person","path":"example.c","language":"c","line":35,"kind":"variable","scope":null}
{"_type":"tag","name":"Instrument","path":"example.c","language":"c","line":29,"kind":"typeAlias","scope":null}
{"_type":"tag","name":"INSTRUMENT_SITAR","path":"example.c","language":"c","line":28,"kind":"enumMember","scope":null}
{"_type":"tag","name":"INSTRUMENT_KEYTAR","path":"example.c","language":"c","line":27,"kind":"enumMember","scope":null}
{"_type":"tag","name":"INSTRUMENT_GUITAR","path":"example.c","language":"c","line":26,"kind":"enumMember","scope":null}
{"_type":"tag","name":"instrument","path":"example.c","language":"c","line":25,"kind":"enum","scope":null}
{"_type":"tag","name":"ANIMAL_ORANGUTAN","path":"example.c","language":"c","line":22,"kind":"enumMember","scope":null}
{"_type":"tag","name":"ANIMAL_GOPHER","path":"example.c","language":"c","line":21,"kind":"enumMember","scope":null}
{"_type":"tag","name":"ANIMAL_GIRAFFE","path":"example.c","language":"c","line":20,"kind":"enumMember","scope":null}
{"_type":"tag","name":"ANIMAL_HORSE","path":"example.c","language":"c","line":19,"kind":"enumMember","scope":null}
{"_type":"tag","name":"ANIMAL_TIGGER","path":"example.c","language":"c","line":18,"kind":"enumMember","scope":null}
{"_type":"tag","name":"ANIMAL_TIGER","path":"example.c","language":"c","line":17,"kind":"enumMember","scope":null}
{"_type":"tag","name":"ANIMAL_TOUCAN","path":"example.c","language":"c","line":16,"kind":"enumMember","scope":null}
{"_type":"tag","name":"animal","path":"example.c","language":"c","line":15,"kind":"enum","scope":null}
{"_type":"tag","name":"RED","path":"example.c","language":"c","line":13,"kind":"enumMember","scope":null}
{"_type":"tag","name":"BLACK","path":"example.c","language":"c","line":13,"kind":"enumMember","scope":null}
{"_type":"tag","name":"sweet_sweet_numbers","path":"example.c","language":"c","line":11,"kind":"variable","scope":null}
{"_type":"tag","name":"age","path":"example.c","language":"c","line":10,"kind":"constant","scope":null}
{"_type":"tag","name":"count2","path":"example.c","language":"c","line":9,"kind":"variable","scope":null}
{"_type":"tag","name":"count","path":"example.c","language":"c","line":8,"kind":"constant","scope":null}
{"_type":"tag","name":"author","path":"example.c","language":"c","line":7,"kind":"constant","scope":null}
{"_type":"tag","name":"name","path":"example.c","language":"c","line":6,"kind":"constant","scope":null}
{"_type":"tag","name":"BUFSIZE","path":"example.c","language":"c","line":4,"kind":"macro","scope":null}

